python/mxnet/rnn/rnn_cell.py

```diff
@@ -906,13 +921,76 @@ def __call__(self, inputs, states):
         next_states = []
         p = 0
         for cell in self._cells:
+            assert type(cell) is not BidirectionalCell
             n = len(cell.state_shape)
             state = states[p:p+n]
             p += n
             inputs, state = cell(inputs, state)
             next_states.append(state)
         return inputs, sum(next_states, [])
 
+    def unroll(self, length, inputs=None, begin_state=None,
+               input_prefix='', layout='NTC', merge_outputs=False):
+        """Unroll an RNN cell across time steps.
+
+        Parameters
+        ----------
+        length : int
+            number of steps to unroll
+        inputs : Symbol, list of Symbol, or None
+            if inputs is a single Symbol (usually the output
+            of Embedding symbol), it should have shape
+            (batch_size, length, ...) if layout == 'NTC',
+            or (length, batch_size, ...) if layout == 'TNC'.
+
+            If inputs is a list of symbols (usually output of
+            previous unroll), they should all have shape
+            (batch_size, ...).
+
+            If inputs is None, Placeholder variables are
+            automatically created.
+        begin_state : nested list of Symbol
+            input states. Created by begin_state()
+            or output state of another cell. Created
+            from begin_state() if None.
+        input_prefix : str
+            prefix for automatically created input
+            placehodlers.
+        layout : str
+            layout of input symbol. Only used if inputs
+            is a single Symbol.
+        merge_outputs : bool
+            if False, return outputs as a list of Symbols.
+            If True, concatenate output across time steps
+            and return a single symbol with shape
+            (batch_size, length, ...) if layout == 'NTC',
+            or (length, batch_size, ...) if layout == 'TNC'.
+
+        Returns
+        -------
+        outputs : list of Symbol
+            output symbols.
+        states : Symbol or nested list of Symbol
+            has the same structure as begin_state()
+        """
+
+        axis = layout.find('T')
+        num_cells = len(self._cells)
+        if begin_state is None:
+            begin_state = self.begin_state()
+
+        p = 0
+        for i, cell in enumerate(self._cells):
+            n = len(cell.state_shape)
+            states = begin_state[p:p+n]
+            p += n
+            inputs, states = cell.unroll(length, inputs=inputs,
+                    input_prefix=input_prefix, begin_state=states,
+                    layout=layout, merge_outputs=(i == num_cells-1 and
```

Comment:
This is inefficient when using multiple fusedrnncells.
extend merge_outputs to True, False, None so that:
True: concat all outputs
False: split all outputs
None: either is fine, use which ever that's faster.